lib/oli/authoring/course.ex

```diff
@@ -338,4 +340,95 @@ defmodule Oli.Authoring.Course do
       description: "New family from project creation"
     }
   end
+
+  def get_project_survey(project_id) do
+    ProjectResource
+    |> join(:inner, [pr], rev in Revision, on: pr.resource_id == rev.resource_id)
+    |> join(:left, [_, rev], rev2 in Revision,
+      on: rev.resource_id == rev2.resource_id and rev2.id > rev.id
+    )
+    |> join(:inner, [pr], p in Project, on: pr.project_id == p.id)
+    |> where(
+      [pr, rev, rev2, p],
+      pr.project_id == ^project_id and
+        rev.resource_id == p.required_survey_resource_id and
+        rev.resource_type_id == ^ResourceType.get_id_by_type("page") and rev.deleted == false and
+        is_nil(rev2)
+    )
+    |> select([_, rev], rev)
+    |> limit(1)
+    |> Repo.one()
+  end
+
+  defp update_project_required_survey_resource_id(project_id, resource_id) do
+    Project
+    |> where([p], p.id == ^project_id)
+    |> Repo.one()
+    |> Project.changeset(%{required_survey_resource_id: resource_id})
+    |> Repo.update()
+  end
+
+  defp remove_survey_for_existing_project_sections(project_id, survey_id) do
```

Comment:
When a course author decides to remove or change the required survey in their project, we do not want to remove or change it from course sections that are using that project.